compiler/aarch64/codegen/OMRMemoryReference.hpp

```diff
@@ -63,6 +63,7 @@ class OMR_EXTENSIBLE MemoryReference : public OMR::MemoryReference
 
    TR::UnresolvedDataSnippet *_unresolvedSnippet;
    TR::SymbolReference *_symbolReference;
+   TR::Register *_extraRegister; // for use with unresolved snippet
```

Comment:
I think it would be useful to document what the extra register is for.  Any downstream project can use it to associate an additional assigned register to a memory reference that can be used for project-specific purposes.  This register is in addition to the base and index registers and isn't directly part of the addressing expression in the memory reference.  An example of what this could be used for is to help synthesize unresolved data reference addresses at runtime.

Reply:
Added comments in the code on the potential use of the extra register.